code/go/0chain.net/chaincore/chain/state/state_context.go

```diff
@@ -227,25 +227,34 @@ func (sc *StateContext) GetSignatureScheme() encryption.SignatureScheme {
 }
 
 func (sc *StateContext) GetTrieNode(key datastore.Key) (util.Serializable, error) {
-	if encryption.IsHash(key) {
-		return nil, common.NewError("failed to get trie node", "key is too short")
+
+	key_hash := encryption.Hash(key)
+
+	if !encryption.IsHash(key_hash) {
```

Comment:
Can this test ever fail? You have just created `key_hash`, so isn't it a hash? Same for insert and delete.